ce_expansion/atomgraph/bcm.py

```diff
@@ -0,0 +1,207 @@
+from ce_expansion.data.gamma import GammaValues
+import numpy as np
+import itertools
+from ce_expansion.data.gamma import GammaValues
+import collections.abc
+import ase.units
+
+
+def recursive_update(d, u):
+    #Function found by team to update Gamma and Bulk Values
+    for k, v in u.items():
+        if isinstance(v, collections.abc.Mapping):
+            d[k] = recursive_update(d.get(k, {}), v)
+        else:
+            d[k] = v
+    return d
+
+
+class BCModel:
+    def __init__(self, atoms, bond_list, metal_types=None):
+        """
+        Based on metal_types, create ce_bulk and gamma dicts from the data given
+
+        Args:
+            atoms: ASE atoms object which contains the data of the NP being tested
+            bond_list: a CSV containing the bond data
+        KArgs:
+            metal_types: List of metals found within the nano-particle
+        """
+
+        self.atoms = atoms
+        self.bond_list = bond_list
+
+        if metal_types is None:
+            # get metal_types from atoms object
+            self.metal_types = sorted(set(atoms.symbols))
+        else:
+            # ensure metal_types to unique, sorted list of metals
+            self.metal_types = sorted(set(metal_types))
+
+        self.metal_pairs = list(itertools.combinations_with_replacement(
+                                    self.metal_types, 2))
```

Comment:
Does this need to be an attribute? It looks like you only use it in a for-loop once. Will it be used elsewhere as a list (i.e. in memory), or can we just call this one-liner when necessary?

If you think it could be used elsewhere, an alternative idea is to create a simple method:

```python
def get_metal_pairs(self):
    return itertools.combinations_with_replacement(self.metal_types, 2))
```

Of course, you don't have to do this, but it's just an idea.

Reply:
One last idea is to just replace your only call to the attribute (in `_get_bcm_params`) with this one-liner. There's also no need to cast it to a list since you're only using it in a for-loop (more efficient to leave it as a generator).